spec/Random.Spec.savi

```diff
@@ -85,3 +85,15 @@
     total U32 = 0
     count.times -> (total += random.unbiased_u32_less_than(52))
     assert: (total.f32 / count.f32) == 25.514451980590820
+
+  :it "generates UUIDs version 4, variant 1"
+    random = _FakeRandom6432.new
```

Comment:
@jemc I'm pulling from `_FakeRandom6432` but I would actually love to test this with `SecureRandom`, too. How could we go about this?

Reply:
@tonchis 
hmmm, I don't think `uuid` belongs to be part of the `Random` library. Think about uuid v4 (timestamp). 

To me, a `Random` library has to deal with generating random values and guarantee certain distributions, which is a very very complex topic on it's own. UUID is a specification (RFC 4122) which deserves it's own library and test suit.

Reply:
@mneumann It's possible that adding a `UUID` library would be useful, but generating a random UUID is so common and so simple (small in scope to implement) that I think it fits nicely here in `Random`. Along with other potential things like generating a random `base64` or `hex` string of a certain number of bytes. Essentially `Random.uuid` is just `hex` of a certain length with some dash characters in the middle. So I think it all fits together nicely, and also matches what you might expect to find if you're familiar with Ruby/Crystal.

@tonchis - I'm curious to hear more about why you want to test `SecureRandom` here - what kinds of bugs are you hoping such tests would detect?

If anything it might be more fruitful to test `DeterministicRandom` instead, since you could give it a certain seed value, and test that the exact same UUID string (or series of UUID strings) was always produced from that seed. `SecureRandom` is harder to use effectively in tests because it is not deterministic.

However, it's definitely possible to use it here. And since you asked how you would go about doing so, I can answer that here:

If you look in the manifest for `Random`, you can see that the test suite already pulls in `DeterministicRandom` as a dependency [here](https://github.com/savi-lang/Random/blob/b9071e51985771f6fc8eb8707e5bdf1eff2bb524/manifest.savi#L8-L10).

You could pull in `SecureRandom` as a dependency in the same way, using a command like `savi deps add SecureRandom --for spec`, which should modify the manifest file and download the dependency. After doing so, it would automatically be available and in scope to refer to in your tests.

Reply:
@jemc I was thinking of testing with `SecureRandom` not precisely because of bugs, but because this way we'll have documenting tests with what will likely be the most common use case for `.uuid`.

However, I like the idea of being able to test the output with `DeterministcRandom` to showcase the uuid.

Reply:
@jemc I got your point. Makes sense to have it easily available as `Random.uuid`.